tests/unit/s2n_send_multirecord_test.c

```diff
@@ -495,5 +495,89 @@ int main(int argc, char **argv)
         EXPECT_TRUE(conn->out.high_water_mark < context.bytes_sent);
     };
 
+    /* Send a post-handshake message when records are buffered, and IO blocks */
+    {
+        s2n_blocked_status blocked = 0;
+
+        DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(conn);
+        EXPECT_OK(s2n_connection_set_secrets(conn));
+        EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+        EXPECT_SUCCESS(s2n_connection_set_send_cb(conn, s2n_test_send_cb));
+
+        /* Find size of a KeyUpdate record */
+        size_t key_update_size = 0;
+        {
+            struct s2n_send_context context = context_all_ok;
+            EXPECT_SUCCESS(s2n_connection_set_send_ctx(conn, (void *) &context));
+
+            conn->key_update_pending = true;
+            EXPECT_SUCCESS(s2n_post_handshake_send(conn, &blocked));
+            EXPECT_FALSE(conn->key_update_pending);
+            key_update_size = context.bytes_sent;
+        }
```

Comment:
nit: `;`

Reply:
It's not a test case and clang-format doesn't seem to be upset. Do you think the ";" is necessary?

Reply:
In general clang-format isn't upset by the lack of a semicolon. It's only in cases where a doubly nested blocks, with on statements between the enclosing block and the enclosed block and where the enclosed block declares a variable as the first thing.

e.g.
```
{
    {
        int variable = 1;
    };
};

```

It thinks this is a C++ initializer list and then tries to put it on one line and gets very confused for the rest of the file.

We decided to get around this issue to adopt a style in which all blocks end in a semicolon which solves the issue.

e.g.
```
{
    {
        int variable = 1;
    };
};
```

Reply:
For future reference: 
The cases that trip clang-format up are only those where we would have nested contexts and no statements in between. In our code that doesn't occur with contexts except for those which are used to isolate test cases. (Because it is a common pattern to isolate groups of test cases for ease of navigation, but that pattern doesn't make sense in other places in the code.)

Therefore the style we adopt is to add these `;`'s on all contexts used to isolate test cases; but not on contexts used to do anything else.